Mathlib/MeasureTheory/MeasurableSpace/Basic.lean

```diff
@@ -1986,30 +1993,110 @@ instance [MeasurableSpace α] [CountablyGenerated α] [MeasurableSpace β] [Coun
+  rcases h.1 with ⟨b, bct, hbm, hb⟩
+  refine' ⟨generateFrom b, _, _, generateFrom_le hbm⟩
+  · use b
+  exact @SeparatesPoints.mk _ (generateFrom b) fun x y hxy ↦ hb _ trivial _ trivial
+    fun s hs ↦ hxy _ $ measurableSet_generateFrom hs
+
 open Classical
 
-/-- If a measurable space is countably generated and separates points, it admits a measurable
-injection into the Cantor space `ℕ → Bool` (equipped with the product sigma algebra). -/
+/-- If a measurable space is countably generated and separates points, it is measure equivalent
+to some some subset of the Cantor space `ℕ → Bool` (equipped with the product sigma algebra). -/
+theorem measurableEquiv_nat_bool_of_countablyGenerated [MeasurableSpace α]
+    [CountablyGenerated α] [SeparatesPoints α] :
+    ∃ s : Set (ℕ → Bool), Nonempty (α ≃ᵐ s) := by
+  rcases exists_seq_generated α with ⟨e, rfl⟩
+  letI := generateFrom (range e)
+  let f : α → ℕ → Bool := (· ∈  e ·)
+  have : Injective f := by
+    intro x y hxy
+    apply separating_of_generateFrom (range e)
+    rintro - ⟨n, rfl⟩
+    rw [← decide_eq_decide]
+    exact congr_fun hxy n
+  use range f, ofInjective _ this
+  · apply Measurable.subtype_mk
+    rw [measurable_pi_iff]
+    refine fun n ↦ measurable_to_bool ?_
+    simp only [preimage, mem_singleton_iff, Bool.decide_iff, setOf_mem_eq]
+    exact measurableSet_generateFrom $ mem_range_self _
+  apply measurable_generateFrom
+  rintro _ ⟨n, rfl⟩
+  rw [← Equiv.image_eq_preimage _ _]
+  refine' ⟨{y | y n}, by measurability, _⟩
+  rw [← Equiv.preimage_eq_iff_eq_image]
+  simp
+
+
+/-- If a measurable space admits a countable sequence of measurable sets separating points,
+it admits a measurable injection into the Cantor space `ℕ → Bool`
+(equipped with the product sigma algebra). -/
 theorem measurable_injection_nat_bool_of_countablyGenerated [MeasurableSpace α]
     [HasCountableSeparatingOn α MeasurableSet univ] :
-    ∃ f : α → ℕ → Bool, Measurable f ∧ Function.Injective f := by
-  rcases exists_seq_separating α MeasurableSet.empty univ with ⟨e, hem, he⟩
-  refine ⟨(· ∈ e ·), ?_, ?_⟩
-  · rw [measurable_pi_iff]
-    refine fun n ↦ measurable_to_bool ?_
-    simpa only [preimage, mem_singleton_iff, Bool.decide_iff, setOf_mem_eq] using hem n
-  · exact fun x y h ↦ he x trivial y trivial fun n ↦ decide_eq_decide.1 <| congr_fun h _
-#align measurable_space.measurable_injection_nat_bool_of_countably_generated MeasurableSpace.measurable_injection_nat_bool_of_countablyGenerated
+    ∃ f : α → ℕ → Bool, Measurable f ∧ Injective f := by
+  rcases exists_countablyGenerated_le_of_hasCountableSeparatingOn α with ⟨m', _, _, m'le⟩
+  rcases @measurableEquiv_nat_bool_of_countablyGenerated _ m' _ _ with ⟨s, ⟨f⟩⟩
+  refine' ⟨Subtype.val ∘ f, _, Subtype.coe_injective.comp (MeasurableEquiv.injective _)⟩
+  apply measurable_subtype_coe.comp
+  exact f.measurable_toFun.mono m'le (by rfl)
+
+variable {α}
+
+theorem measurableSingletonClass_of_hasCountableSeparatingPoints
```

Comment:
This should be an instance. However, making it one, even with priority 1, causes timeouts in in BorelSpace/Basic.lean